bluemira/base/components.py

```diff
@@ -107,21 +108,74 @@ def get_component(
 
         Notes
         -----
-            This function is just a wrapper of the anytree.search.findall_by_attr
+            This function is just a wrapper of the anytree.search.findall
             function.
         """
-        if full_tree:
-            found_components = anytree.search.findall_by_attr(self.root, name)
-        else:
-            found_components = anytree.search.findall_by_attr(self, name)
+        return self._get_thing(
+            lambda n: anytree.search._filter_by_name(n, "name", name), first, full_tree
+        )
 
-        if len(found_components) == 0:
-            return None
+    def get_component_properties(
+        self, properties: Union[Iterable[str], str], first=True, full_tree=False
+    ):
+        """
+        Get properties from a component
+
+        Parameters
+        ----------
+        name: str
+            The name of the component to search for.
+        first: bool
+            If True, only the first element is returned, by default True.
+        full_tree: bool
+            If True, searches the tree from the root, else searches from this node, by
+            default False.
+
+        Returns
+        -------
+        Lists of properties of components
+
+        Notes
+        -----
+            This function is just a wrapper of the anytree.search.findall or find
+            functions.
+        """
+        if isinstance(properties, str):
+            properties = [properties]
+
+        def filter_(node, properties):
+            return all(hasattr(node, prop) for prop in properties)
+
+        found_nodes = self._get_thing(lambda n: filter_(n, properties), first, full_tree)
+
+        if found_nodes is None:
+            return tuple([] for _ in properties)
 
-        if first:
-            return found_components[0]
+        if not isinstance(found_nodes, Iterable):
+            if len(properties) == 1:
+                return getattr(found_nodes, properties[0])
+            return [getattr(found_nodes, prop) for prop in properties]
+        else:
+            return tuple(
+                obj.tolist()
+                for obj in np.array(
+                    [
+                        [getattr(node, prop) for prop in properties]
+                        for node in found_nodes
+                    ],
+                    dtype=object,
+                ).T
+            )
```

Comment:
If you have a nicer way to 'rotate' a list of lists I'm open to it :)

Reply:
How about:

```python
node_properties = [
    [getattr(node, prop) for prop in properties] for node in found_nodes
]
return tuple(map(list, zip(*node_properties)))
```

Either way, I think this is a bit cryptic. Probably warrants a comment noting that we're collecting similar _property_ values, instead of ordering by _node_ property values

Reply:
Thats nicer :) Sure I'll add both